src/compiler/checker.ts

```diff
@@ -20885,6 +20885,17 @@ namespace ts {
                         }
                     }
                 }
+                else if (/* !catchClause.variableDeclaration && */ languageVersion < ScriptTarget.ESNext) {
+                    const blockLocals = catchClause.block.locals;
+                    if (blockLocals) {
+                        forEachKey(blockLocals, caughtName => {

```

Comment:
I removed this block of change. Since we don't need check and error on a named variable anymore.